pywbem/cim_operations.py

```diff
@@ -643,6 +661,19 @@ def __init__(self, url, creds=None, default_namespace=DEFAULT_NAMESPACE,
             Note that not all situations can be handled within this timeout, so
             for some issues, operations may take longer before raising an
             exception.
+
+          use_pull_operations (:class:`py:bool`):
+            Allows the user to force the use of either only the pull operations
+            or only the non-pull operations for the Iter... methods. Normally
+            this should NOT be used but it may prove useful in some test or
+            error cases.
+            If `None` ignored and the server sets the attribute based on the
+            response to the first pull operation.
+            If `True` forces the client to use only the pull operations for
+            the Iter... methods and return any response  to a pull operation
+            including `CIM_ERR_NOT_SUPPORTED` as an error on this connection.
+            If `False` forces the client to use only the non-pull operations
+            (ex. EnumerateInstances) for the Iter... methods.
```

Comment:
Comments on the description of the `use_pull_operation`  ctor argument:

1. The "force" only works in one direction, at least in the current implementation: `False` forces non-use even if pull is supported by the server, but `True` specifies that an attempt to use pull is made, which may fall back to using non-pull.

2. On "Normally this should not be used...": It is not clear what the "this" refers to.

3. For `None`, I would like to see a more explicit statement that pull is attempted.

4. For `True`, it does not force the use of only pull. That may have been an a prior implementation, but the current code attempts pull and if it fails with not supported, goes on to attempt non-pull. From a perspective of the (unadjusted) input argument, `None` specifies exactly the same as `True`. Which is fine IMO, just the description needs to be fixed.

5. On "If `False` forces ...": I don't like this super-short form. Better use "`False` forces ...".

How about this description:

```
            Controls the use of pull operations in any `Iter...()` methods. The default
            value will work fine on any WBEM server whether or not it supports pull
            operations.
            `None` and `True` mean that pull operations will be attempted, and if
            the WBEM server does not support pull operations, the corresponding
            instance variable is set to `False`, causing the traditional operations
            to be used in the future.
            `False` means that only the traditional operations will be used,
            regardless of whether or not the WBEM server also supports
            pull operations.
```